pkg/cvo/metrics_test.go

```diff
@@ -373,13 +386,72 @@ func Test_operatorMetrics_Collect(t *testing.T) {
 				},
 			},
 			wants: func(t *testing.T, metrics []prometheus.Metric) {
-				if len(metrics) != 4 {
+				if len(metrics) != 5 {
 					t.Fatalf("Unexpected metrics %s", spew.Sdump(metrics))
 				}
 				expectMetric(t, metrics[0], 2, map[string]string{"type": "initial", "version": "", "image": "", "from_version": ""})
 				expectMetric(t, metrics[1], 2, map[string]string{"type": "cluster", "version": "0.0.2", "image": "test/image:1", "from_version": ""})
 				expectMetric(t, metrics[2], 0, map[string]string{"type": "failure", "version": "0.0.2", "image": "test/image:1", "from_version": ""})
 				expectMetric(t, metrics[3], 0, map[string]string{"type": "current", "version": "0.0.2", "image": "test/image:1", "from_version": ""})
+				expectMetric(t, metrics[4], 0, map[string]string{"type": ""})
+			},
+		},
+		{
+			name: "collects openshift-install info",
+			optr: &Operator{
+				releaseVersion: "0.0.2",
+				releaseImage:   "test/image:1",
+				releaseCreated: time.Unix(3, 0),
+				cmLister: &cmLister{
+					Items: []*corev1.ConfigMap{{
+						ObjectMeta: metav1.ObjectMeta{Name: "openshift-install"},
+						Data: map[string]string {"version": "v0.0.2", "invoker": "jane"},
+					}},
+				},
+			},
+			wants: func(t *testing.T, metrics []prometheus.Metric) {
+				if len(metrics) != 2 {
+					t.Fatalf("Unexpected metrics %s", spew.Sdump(metrics))
+				}
+				expectMetric(t, metrics[0], 3, map[string]string{"type": "current", "version": "0.0.2", "image": "test/image:1"})
+				expectMetric(t, metrics[1], 0, map[string]string{"type": "openshift-install", "version": "v0.0.2", "invoker": "jane"})
+			},
+		},
+		{
+			name: "collects empty openshift-install info",
+			optr: &Operator{
+				releaseVersion: "0.0.2",
+				releaseImage:   "test/image:1",
+				releaseCreated: time.Unix(3, 0),
+				cmLister: &cmLister{
+					Items: []*corev1.ConfigMap{{
+						ObjectMeta: metav1.ObjectMeta{Name: "openshift-install"},
+					}},
+				},
+			},
+			wants: func(t *testing.T, metrics []prometheus.Metric) {
+				if len(metrics) != 2 {
+					t.Fatalf("Unexpected metrics %s", spew.Sdump(metrics))
+				}
+				expectMetric(t, metrics[0], 3, map[string]string{"type": "current", "version": "0.0.2", "image": "test/image:1"})
+				expectMetric(t, metrics[1], 0, map[string]string{"type": "openshift-install", "version": "<missing>", "invoker": "<missing>"})
```

Comment:
This is somewhat abnormal, usually you'd just return empty values instead of missing.

Reply:
Hmm, my thinking was that empty values were valid (e.g. a UPI install has the installer type, version, and invoker set to the empty value) whereas `<missing>` indicates an anomalous state. If the openshift-install ConfigMap exists, it will have all of those fields specified unless something were to erroneously remove them.

If we change this to use the empty value, then I need to update what's reported in UPI and it's not clear to me what those should be. I could say something like `m.clusterInstaller.WithLabelValues("human", "", "")`, but I didn't want to complicate things if we ever decide to have the installer explicitly call out when a UPI installation was performed (v.s. inferring it based on the absence of state).

Reply:
If you have a good reason, that's ok.

Reply:
All good stuff for the commit message :wink: